File: d04a.c
#include <stdio.h>
#include <string.h>
#include <openssl/md5.h>
int main(int argc,char **argv) {
  char s2[20];
  unsigned char s[16];
  int i=0;
  strcpy(s2,"yzbqklnj");
  do {
    i++;
    sprintf(s2+8,"%d",i);
    MD5((unsigned char *)s2,strlen(s2),s);
  } while (*s || *(s+1) || (*(s+2) & 0xf0));  
  printf("5 zeros =%d\n",i);
  do {
    i++;
    sprintf(s2+8,"%d",i);
    MD5((unsigned char *)s2,strlen(s2),s);
  } while (*s || *(s+1) || (*(s+2)));  
  printf("6 zeros =%d\n",i);
}

File: d04a.php
#!/usr/bin/php
<?php
  $n=1;
  while (strncmp(md5('yzbqklnj'.$n),'00000',5)) $n++;
  echo "5 zeros =$n\n";
  while (strncmp(md5('yzbqklnj'.$n),'000000',6)) $n++;
  echo "6 zeros =$n\n";
?>